client/state/utils.js

```diff
@@ -19,3 +23,34 @@ export function isValidStateWithSchema( state, schema, checkForCycles = false, b
 	}
 	return result.valid;
 }
+
+export function createReducer( initialState = null, handlers = {}, schema = null ) {
+	const defaultHandlers = {
+		[SERIALIZE]: ( state ) => {
+			if ( schema !== null ) {
+				return state;
+			}
+
+			return initialState;
+		},
+		[DESERIALIZE]: ( state ) => {
+			if ( schema !== null && isValidStateWithSchema( state, schema ) ) {
+				return state;
+			}
+
+			return initialState;
+		}
+	};
+
+	handlers = Object.assign( {}, defaultHandlers, handlers );
```

Comment:
Ah, I guess what your original suggestion enables is not serializing state unless a schema is defined? If so, that seems a nice developer convenience for avoiding potential issues.